cns/restserver/internalapi_test.go

```diff
@@ -581,28 +581,28 @@ func validateNCStateAfterReconcile(t *testing.T, ncRequest *cns.CreateNetworkCon
 		t.Fatalf("Unexpected assigned pods, actual: %d, expected: %d", len(svc.PodIPIDByPodInterfaceKey), len(expectedAssignedPods))
 	}
 
-	for ipaddress, podInfo := range expectedAssignedPods {
-		ipId := svc.PodIPIDByPodInterfaceKey[podInfo.Key()]
-		ipConfigstate := svc.PodIPConfigState[ipId[0]]
-
-		if ipConfigstate.GetState() != types.Assigned {
-			t.Fatalf("IpAddress %s is not marked as assigned to Pod: %+v, ipState: %+v", ipaddress, podInfo, ipConfigstate)
-		}
+	for ipaddress, podInfo := range expectedAssignedPods {	
+		for _, ipID := range svc.PodIPIDByPodInterfaceKey[podInfo.Key()] {
+			ipConfigstate := svc.PodIPConfigState[ipID]
+			if ipConfigstate.GetState() != types.Assigned {
+				t.Fatalf("IpAddress %s is not marked as assigned to Pod: %+v, ipState: %+v", ipaddress, podInfo, ipConfigstate)
+			}
 
-		// Validate if IPAddress matches
-		if ipConfigstate.IPAddress != ipaddress {
-			t.Fatalf("IpAddress %s is not same, for Pod: %+v, actual ipState: %+v", ipaddress, podInfo, ipConfigstate)
-		}
+			// Validate if IPAddress matches
```

Comment:
This section got indented because I had to add an extra for loop for ranging over svc.PodIPIDByPodInterfaceKey[podInfo.Key()]  since the map's value is now a slice of IPs instead of just an IP.